clippy_lints/src/methods/mod.rs

```diff
@@ -3261,6 +3262,43 @@ declare_clippy_lint! {
     "calling `drain` in order to `clear` a container"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for calls to `Path::join` that start with a path separator, like `\\` or `/`..
+    ///
+    /// ### Why is this bad?
+    /// `.join()` arguments starting with a separator  (`/` or `\\`) can replace the entire path.
+    /// If this is intentional, prefer using `Path::new()` instead.
+    ///
+    /// See [`Path::join()`](https://doc.rust-lang.org/std/path/struct.Path.html#method.join)
+    ///
+    /// ### Example
+    /// ```rust
+    /// # use std::path::{Path, PathBuf};
+    /// let path = Path::new("/bin");
+    /// let joined_path = path.join("/sh");
+    /// assert_eq!(joined_path, PathBuf::from("/sh"));
+    /// ```
+    ///
+    /// Use instead;
+    /// ```rust
+    /// # use std::path::{Path, PathBuf};
+    /// let path = Path::new("/bin");
+    ///
+    /// // If this was unintentional, remove the leading separator
+    /// let joined_path = path.join("sh");
+    /// assert_eq!(joined_path, PathBuf::from("/bin/sh"));
+    ///
+    /// // If this was intentional, create a new path instead
+    /// let new = Path::new("/sh");
+    /// assert_eq!(new, PathBuf::from("/sh"));
+    /// ```
+    #[clippy::version = "1.70.0"]
+    pub JOIN_ABSOLUTE_PATHS,
+    pedantic,
```

Comment:
Is there a reason this is `pedantic`? I think there's an argument to be had for this to be `suspicious` or `correctness` instead, so it's enabled-by-default.

Reply:
I actually didn't know there was correctness as an option. I went with pedantic as it doesn't seem like a breaking mistake, but correctness makes way more sense here. I'm choosing that.